### DISCUSSION TITLE
How to Dynamically Update Table Data on Edit Button Click in Filament (Livewire) from Details to Subdetails?

### DISCUSSION THREAD

**Discussion by itbaracoal** [filamentphp/filament#14884] [Help]
### Package

Table builder

### Package Version

v3.2

### How can we help you?

Hello,
I'm working on a Filament + Livewire project and facing some challenges related to dynamically updating a table based on data passed through edit actions. Here's what I'm trying to achieve:

![image](https://github.com/user-attachments/assets/c5e965f5-8752-4a6b-9b26-1eacf247351a)

- Edit Button Triggering Table Update: I have an Edit button in my table. When the user clicks the Edit button for a particular record, I want to:

Update the detail_id and lahan_id in the Livewire component.
Refresh the table to reflect the new data based on these updated IDs.
Currently, I'm passing detail_id and lahan_id as props to the component. The problem is that when the button is clicked, the table doesn’t automatically reflect the updated data. How can I trigger a re-render or refresh of the table when the Edit button is clicked and show all related dokumen?

this is current codes
// Code for passing detail_id and lahan_id to the Livewire component (Comp1)
`<x-filament::button wire:click="editModalDetail({{ $detail['id'] ?? 'null' }}, {{ $detail['lahan_id'] ?? 'null' }})">
    Edit
</x-filament::button>`

// Code inside Livewire component (Comp1)
```
`public function editModalDetail($detail_id, $lahan_id)
{
    $this->detail_id = $detail_id;
    $this->lahan_id = $lahan_id;

    // Trigger the table to refresh
    $this->emit('refreshTable');
}`
```

```
`<div class="col-span-1">
                    @livewire('tab.dokumen-verifikasi-lahan', [
                        'record' => $this->record,
                        'detail_id' => $this->detail_id,
                        'lahan_id' => $this->lahan_id,
                    ])
</div>`
```
                
 ------- Component 2
```
 `public function mount(ProsesTpl $record, $detail_id, $lahan_id)
    {
        $this->record = $record;
        $this->detail_id = $detail_id;
        $this->lahan_id = $lahan_id;
    }
    public function table(Table $table): Table
    {
        return $table
            ->heading('Dokumen Verifikasi')
            ->query(MstLahanVerifikasiDokumen::where('lahan_id', $this->lahan_id)
                ->with('proyekDokumen.dokumenLegalitas'))
            ->columns([
                Tables\Columns\TextColumn::make('review_legal_tgl')->label('Tgl Review Legal')->date(format: 'd/m/y'),
                Tables\Columns\TextColumn::make('final_review_legal_tgl')->label('Tgl Final Review Legal')->date(format: 'd/m/y'),
                Tables\Columns\TextColumn::make('check_list_dokumen')->label('Check List Dokumen'),
                Tables\Columns\TextColumn::make('final_review_legal_desc')->label('Final Review Legal Desc')->limit(50),
                Tables\Columns\TextColumn::make('tanggapan_tpl_desc')->label('Tanggapan TPL Desc')->limit(50),
                Tables\Columns\TextColumn::make('keterangan')->label('Keterangan')->limit(50),
            ])
            ->filters([`
```
            
            
```
  `<div>
    {{ $this->table }}
</div>`
```

I would appreciate any insights, solutions, or best practices to achieve this in Filament and Livewire.

Thanks in advance ☺

**Comment by leandrocfe** [comment]
Maybe  [resetTable](https://filamentphp.com/docs/3.x/tables/advanced/#resetting-the-table)?
